### PR TITLE
deps(ui): Upgrade to stylelint v14

### PR DESCRIPTION
**Pull request by scttcper**
v13 is no longer supported by the [stylelint vscode extension](https://github.com/stylelint/vscode-stylelint/releases/tag/v1.0.0).
- removes `stylelint-config-styled-components`
- adds and `postcss-syntax` and `postcss-jsx`

`stylelint-processor-styled-components` is also dead
https://github.com/styled-components/stylelint-processor-styled-components

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -176,16 +176,17 @@
...
     "react-refresh": "0.11.0",
     "react-select-event": "5.3.0",
     "react-test-renderer": "17.0.2",
     "size-limit": "^5.0.5",
     "speed-measure-webpack-plugin": "^1.5.0",
     "storybook-dark-mode": "^1.0.8",
-    "stylelint": "13.13.1",
-    "stylelint-config-prettier": "^8.0.2",
-    "stylelint-config-recommended": "^5.0.0",
-    "stylelint-config-styled-components": "^0.1.1",
```
Comment (billyvg) [inline package.json:188]: Does this mean stylelint no longer lints styles in styled components?

Comment (scttcper) [reply package.json:188]: from my understanding that package was to fix something temporarily https://github.com/styled-components/stylelint-config-styled-components#why

Comment (scttcper) [reply package.json:188]: i guess this is all it does? https://github.com/styled-components/stylelint-config-styled-components/blob/master/index.js

**Comment by scttcper** [comment]
we might be able to remove `stylelint-processor-styled-components` as well

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**